SCons/Platform/win32.py

```diff
@@ -321,7 +322,7 @@ def get_architecture(arch=None):
         arch = os.environ.get('PROCESSOR_ARCHITEW6432')
         if not arch:
             arch = os.environ.get('PROCESSOR_ARCHITECTURE')
-    return SupportedArchitectureMap.get(arch, ArchDefinition('', ['']))
+    return SupportedArchitectureMap.get(arch, ArchDefinition(platform.machine(), [platform.machine()]))
```

Comment:
Hmmm, I dont think we want to have the default be an `ArchDefinition` instance, since the return should be a string. Alternate proposal:
```python
def get_architecture(arch=None) -> str:
    """Returns the definition for the specified architecture string.

    If no string is specified, the system default is returned (as defined
    by the PROCESSOR_ARCHITEW6432 or PROCESSOR_ARCHITECTURE environment
    variables).
    """
    if not arch:
        arch = os.environ.get('PROCESSOR_ARCHITEW6432')
    if not arch:
        arch = os.environ.get('PROCESSOR_ARCHITECTURE')
    if not arch:
        # we should still try to have some kind of value - the unit tests
        # try each platform in turn, and the env vars won't be set if
        # this isn't "natively" a win32 system.
        arch = platform.machine()
    return SupportedArchitectureMap.get(arch, arch)
```

Reply:
I think you're mistaken:

```
SCons/Platform/win32.py:def get_architecture(arch=None):
SCons/Platform/win32.py:    env['HOST_ARCH']      = get_architecture().arch
SCons/Platform/os2.py:    env['HOST_ARCH']      = win32.get_architecture().arch
```

Reply:
and so it does, I missed that's how they use it.  That extra bit of complexity doesn't make any sense to me, it builds a dict mapping all the arch permutations not to the wanted string, but to `ArchDefinition` instances, when the only two consumers of this then immediately look up the single string anyway and ignore anything else... but what the heck, it works I guess.  Objection withdrawn.